src/lib/collections/__init__.js

```diff
@@ -301,10 +301,195 @@ var $builtinmodule = function(name)
 
 
     // OrderedDict
-    mod.OrderedDict = function OrderedDict(items) {
-        throw new Sk.builtin.NotImplementedError("OrderedDict is not implemented")
+    mod.OrderedDict = function OrderedDict(items)
+    {
+        if (!(this instanceof mod.OrderedDict))
+        {
+            return new mod.OrderedDict(items);
+        }
+
+        this.orderedkeys = [];
+
+        Sk.builtin.dict.call(this, items);
+
+        return this;
+    }
+
+    mod.OrderedDict.prototype = Object.create(Sk.builtin.dict.prototype);
+
+    mod.OrderedDict.prototype.tp$name = 'OrderedDict'
+
+    mod.OrderedDict.prototype.ob$type = Sk.builtin.type.makeIntoTypeObj('OrderedDict', mod.OrderedDict);
+
+    mod.OrderedDict.prototype['$r'] = function()
+    {
+        var v;
+        var iter, k;
+        var ret = [];
+        var pairstr;
+        for (iter = this.tp$iter(), k = iter.tp$iternext();
+             k !== undefined;
+             k = iter.tp$iternext()) {
+            v = this.mp$subscript(k);
+            if (v === undefined) {
+                //print(k, "had undefined v");
+                v = null;
+            }
+            ret.push("(" + Sk.misceval.objectRepr(k).v + ", " + Sk.misceval.objectRepr(v).v + ")");
+        }
+        pairstr = ret.join(", ");
+        if (ret.length > 0)
+        {
+            pairstr = "[" + pairstr + "]";
+        }
+        return new Sk.builtin.str("OrderedDict(" + pairstr + ")");
+    }
+
+    mod.OrderedDict.prototype.mp$ass_subscript = function(key, w)
+    {
+        var idx = this.orderedkeys.indexOf(key);
+        if (idx == -1)
+        {
+            this.orderedkeys.push(key);
+        }
+
+        return Sk.builtin.dict.prototype.mp$ass_subscript.call(this, key, w);
+    }
+
+    mod.OrderedDict.prototype.mp$del_subscript = function(key, w)
+    {
+        var idx = this.orderedkeys.indexOf(key);
+        if (idx != -1)
+        {
+            this.orderedkeys.splice(idx, 1);
+        }
+
+        return Sk.builtin.dict.prototype.mp$del_subscript.call(this, key, w);
```

Comment:
Why do you pass here the w parameter? I've added checks on the dict object, that checks the number of parameters. ```del_subscript``` only expects the key and the third ```w``` is not used at all.